appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java

```diff
@@ -1264,6 +1266,96 @@ public void DrawLine(int x1, int y1, int x2, int y2) {
     view.invalidate();
   }
 
+  /**
+   * Draws a shape on the canvas, 
+   * point list should be a list contains sub-lists with two number which represents a coordinate
+   *
+   * @param pointList  a list of points, should contains sub-lists with two number which represents a coordinate
+   * @param fill  true for filled shape; false for shape outline
+   */
+  @SimpleFunction
+  public void DrawShape(YailList pointList, boolean fill) {
+    Path path = parsePath(parsePointList(pointList));
+    if (path == null) {
+      $form().dispatchErrorOccurredEvent(this, "DrawShape", ErrorMessages.ERROR_CANVAS_DRAW_SHAPE_BAD_ARGUMENT);
+      return;
+    }
+    path.close();
+    Paint p = new Paint(paint);
+    p.setStyle(fill ? Paint.Style.FILL : Paint.Style.STROKE);
+    view.canvas.drawPath(path, p);
+    view.invalidate();
+  }
+
+  private Path parsePath(float[][] points) {
+    if (points == null) {
+      return null;
+    }
+    float scalingFactor = $form().deviceDensity();
+
+    Path path = new Path();
+    path.moveTo(points[0][0] * scalingFactor, points[0][1] * scalingFactor);
+    for (int i = 1; i < points.length; i++) {
+      path.lineTo(points[i][0] * scalingFactor, points[i][1] * scalingFactor);
+    }
+
+    return path;
+  }
+
+  private float[][] parsePointList(YailList pointList) {
+    if (pointList == null) {
+      return null;
+    }
+    float[][] points = new float[pointList.size()][2];
+    int index = 0;
+    YailList pointYailList;
+    for (Object pointObject : pointList.toArray()) {
+      if (pointObject instanceof YailList) {
+        pointYailList = (YailList) pointObject;
+        if (pointYailList.size() == 2) {
+          try {
+            points[index][0] = Float.parseFloat(pointYailList.getString(0));
+            points[index][1] = Float.parseFloat(pointYailList.getString(1));
+            index++;
+            continue;
+          } catch (NullPointerException ok) {
+          } catch (NumberFormatException ok) {
+          }
+        }
+      }
+      // if continue is not trigered, null will be executed
+      return null;
+    }
+    return points;
+  }
+
+  /**
+   * Draw an arc on Canvas, by drawing a arc from a specified oval
+   *   (specified by left, top, right & bottom),
+   * start angle is 0 when heading to the right, and increase when rotate clockwise.
```

Comment:
Because the users don't see the param annotations below, include a description of `useCenter` and `fill` here.